### DISCUSSION TITLE
Backup and restore?

### DISCUSSION THREAD

**Discussion by ciscojunkiebt** [ESPresense/ESPresense#1245] [Q&A]
I've searched quite a bit but can't find any answer (or question, for that matter), but is there a way to backup/restore the ESPs? Possibly dumping a local YAML/config file from the device? It seems simple enough to setup and get going but it would be nice to dump the config and store it in case any of my ESPs die, then just replace it and dump the config back to the new one.

**Comment by danielgoepp** [comment]
I had this same question, so I'm curious what others know.

There is already a web server running on all these nodes, so I agree it would seem pretty easy to have a GET / POST path with JSON/YAML to backup and restore.

Also, most of the core settings are posted to MQTT (just not advanced settings), so it would be pretty straightforward to write a script to subscribe and save. Also, you can set settings globally via MQTT with the "*" room. What I noticed though is that rooms are not retained so if your MQTT server was reset, these settings wouldn't be available again until all devices are reset and pushed their configurations again.

**Comment by Terastar-Paperclip** [comment]
Tech Lead here. Checked `HttpWebServer.cpp` — this is partially already there.

Per-device overrides (aliases, calibrated RSSI) already have a JSON API today: `GET /json/configs` dumps them, `POST /json` (with a `configs` body) and `DELETE /json/configs?id=...` restore/remove them. No UI button for it yet, but it's scriptable right now if you want to back that part up.

What's *not* covered: full node settings — WiFi, MQTT broker, room name, LED/button/switch pin config. I don't see an existing single-shot export/import route for those (they go through `HeadlessWiFiSettings`/NVS, not the JSON config API). That's a real gap and a reasonable ask given two of you want it, but it's not something I'm picking up myself right now — filing it as **help wanted**: extend the existing `/json/configs` pattern with a `/json/settings` dump+restore route. Reasonably scoped for a contributor.

**Comment by Terastar-Paperclip** [comment]
Filed as #2493.
